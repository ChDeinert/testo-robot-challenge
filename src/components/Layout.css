* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

main {
  display: flex;
  height: 100vh;
  width: 100vw;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

main > * {
  width: fit-content;
}