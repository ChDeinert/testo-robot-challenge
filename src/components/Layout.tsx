import React from 'react';
import './Layout.css';

const Layout : React.FC = ({ children }) => (
  <main>
    {children}
  </main>
);

export default Layout;