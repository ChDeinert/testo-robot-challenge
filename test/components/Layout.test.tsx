import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Layout from '../../src/components/Layout';

describe('<Layout>', () => {
  it('renders a main tag', () => {
    const { container } = render(<Layout />);
    expect(container.querySelector('main')).toBeInTheDocument();
  });

  it('renders it\'s children inside a main tag', () => {
    const { container, getByTestId } = render(<Layout><div data-testid="testChild">testChild</div></Layout>);
    const layoutChild = getByTestId('testChild');
    expect(layoutChild).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});