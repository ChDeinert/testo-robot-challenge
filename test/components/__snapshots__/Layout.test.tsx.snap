// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> renders it's children inside a main tag 1`] = `
<div>
  <main>
    <div
      data-testid="testChild"
    >
      testChild
    </div>
  </main>
</div>
`;
